### PR TITLE
Tem Implementation with tests

### PR DESCRIPTION
**Pull request by arjunsingh3600**
This is the TEM wrapper implementation without the run() and get_image_data() functions (they require a tem sim installation to run).
This PR is blocked by -

- https://github.com/compSPI/ioSPI/pull/24
- https://github.com/compSPI/simSPI/pull/19

@jedyeo @geoffwoollard @thisTyler

### CODE REVIEW

**Comment by jedyeo** [comment]
@ninamiolane @geoffwoollard @fredericpoitevin to clarify this branch holds the implementation for the TEM Simulator Wrapper as well as associated tests (as Arjun stated above). It's blocked by:

> * [Cryoemio ioSPI#24](https://github.com/compSPI/ioSPI/pull/24)
> * [Feature: adding helper functions from SLAC #19](https://github.com/compSPI/simSPI/pull/19)

These contain helper functions that the simulator rely on and as such the builds are failing here.

As an aside, this PR contains some files that should not be in here and we'll work to move them tomorrow morning.

File: simSPI/crd.py
Comment (geoffwoollard) [inline simSPI/crd.py:84]: range(n_particles + 1)

File: simSPI/fov.py
Comment (geoffwoollard) [inline simSPI/fov.py:1]: write field of view instead of FOV here

File: simSPI/fov.py
Comment (geoffwoollard) [inline simSPI/fov.py:169]: no caps. fov_lx etc

File: simSPI/tem.py
Comment (geoffwoollard) [inline simSPI/tem.py:79]: put units in this to make it more readable

File: tests/test_files/sim_config.yaml
Comment (geoffwoollard) [inline tests/test_files/sim_config.yaml:1]: include units in variable names, and incorporate comment info if not too verbose.

File: tests/test_fov.py
Comment (geoffwoollard) [inline tests/test_fov.py:13]: yes, very good! just inline a snippet of data. be aware the pdb data is very silly and all the spaces between columns is uber critical. that's one reason why it was deprecated. so be sure to copy and paste EXACTLY from a real pdb file to avoid column offset

File: tests/test_tem.py
Comment (geoffwoollard) [inline tests/test_tem.py:18]: better to use longer name and not be compact. if it's longer then it's easier to search and replace. just "t" would appear all over the place and then you have to do things manually

**Review by geoffwoollard — CHANGES_REQUESTED** [review]
Good tests! 

Change variable names to include units.

**Comment by jedyeo** [comment]
to clarify the helper functions should not be in this branch and are to be moved out. They are in #19.

**Comment by geoffwoollard** [comment]
If the hooks aren't changing this for you automatically, you can run black locally with a flag for the max line length (on the compSPI/simSPI`.github/workflows` it is 88 as per the error below).

Note that when you commit, and the hook changes something, you have to git add it again and then (try to) commit again.

```
Run $CONDA/bin/flake8 simSPI tests
tests/test_tem.py:26:54: BLK100 Black would make changes.
tests/test_tem.py:26:89: E501 line too long (97 > 88 characters)
tests/test_tem.py:27:89: E501 line too long (97 > 88 characters)
tests/test_tem.py:28:89: E501 line too long (97 > 88 characters)
tests/test_tem.py:29:89: E501 line too long (97 > 88 characters)
tests/test_tem.py:30:89: E501 line too long (95 > 88 characters)
tests/test_tem.py:31:89: E501 line too long (92 > 88 characters)
Error: Process completed with exit code 1.
```

**Comment by arjunsingh3600** [comment]
The tests in this branch are dependant on #42. Will mark PR as ready to review after #42 is merged.

**Comment by geoffwoollard** [comment]
> The tests in this branch are dependant on #42. Will mark PR as ready to review after #42 is merged. Note: This PR contains methods of the tem_wrapper that are not dependant on a local installation. The tests in the PR are supposed to pass. Following the merge of this PR, Another PR will be created for the remaining methods of the tem_wrapper.

Ok I merged https://github.com/compSPI/simSPI/pull/42

**Comment by arjunsingh3600** [comment]
@geoffwoollard Thank you for the comments, they should now be incorporated

File: tests/test_tem.py
Comment (geoffwoollard) [inline tests/test_tem.py:122]: does `voltage_kV` have to match the non capitalization in https://github.com/compSPI/simSPI/blob/tem_with_tests/tests/test_files/sim_config.yaml#L12

also there might be a mismatch in `energy_spread_V`

maybe also change `detector_Nx_px`, `detector_Nx_py`, `detector_Q_efficiency`, `MTF_params` to have no caps

**Review by geoffwoollard — COMMENTED** [review]
it's passing all tests, and the test coverage is 100%, but check that there aren't issues with capitalization consistency. 

besides this seems good to merge to me.

File: tests/test_tem.py
Comment (thisTyler) [reply tests/test_tem.py]: Changes made as suggested - the changes to detector_nx_px and co. are for the best (the capitals were a carryover from imported code) but they also required changes to cryoemio in ioSPI. A new branch "parameter_name_changes" has been opened in ioSPI to this end, so please take a look at https://github.com/compSPI/ioSPI/pull/27 so that these changes can be imported.

Comment (thisTyler) [reply tests/test_tem.py]: Note: this is also why the tests are failing.

**Pull request merged**
